core/templates/dev/head/domain/skill/SkillObjectFactory.js

```diff
@@ -0,0 +1,149 @@
+        misconceptions: this._misconceptions.map(function(misconception) {
+          return misconception.toBackendDict();
+        }),
+        skill_contents: this._conceptCard.toBackendDict(),
+        language_code: this._languageCode,
+        version: this._version
+      };
+    };
+
+    Skill.prototype.copyFromSkill = function(skill) {
+      this._id = skill._id;
+      this._description = skill._description;
+      this._misconceptions = skill._misconceptions;
+      this._conceptCard = skill._conceptCard;
+      this._languageCode = skill._languageCode;
+      this._version = skill._version;
+    };
+
+    Skill.createFromBackendDict = function(skillBackendDict) {
+      return new Skill(
+        skillBackendDict.id,
+        skillBackendDict.description,
+        generateMisconceptionsFromBackendDict(skillBackendDict.misconceptions),
+        ConceptCardObjectFactory.createFromBackendDict(
+          skillBackendDict.skill_contents),
+        skillBackendDict.language_code,
+        skillBackendDict.version);
+    };
+
+    Skill.createEmptySkill = function() {
+      return new Skill(null, null, [], null, null);
+    };
+
+    var generateMisconceptionsFromBackendDict = function(
+        misconceptionsBackendDicts) {
+      return misconceptionsBackendDicts.map(function(
+          misconceptionsBackendDict) {
+        return MisconceptionObjectFactory.createFromBackendDict(
+          misconceptionsBackendDict);
+      });
+    };
+
+    Skill.prototype.setDescription = function(description) {
+      this._description = description;
+    };
+
+    Skill.prototype.getDescription = function() {
+      return this._description;
+    };
+
+    Skill.prototype.getId = function() {
+      return this._id;
+    };
+
+    Skill.prototype.getConceptCard = function() {
+      return this._conceptCard;
+    };
+
+    Skill.prototype.getMisconceptions = function() {
+      return this._misconceptions;
+    };
+
+    Skill.prototype.appendMisconception = function(newMisconception) {
+      this._misconceptions.push(newMisconception);
+    };
+
+    Skill.prototype.removeLastMisconception = function() {
+      this._misconceptions.pop();
+    };
+
+    Skill.prototype.getLanguageCode = function() {
+      return this._languageCode;
+    };
+
+    Skill.prototype.getVersion = function() {
+      return this._version;
+    };
+
+    Skill.prototype.getNextMisconceptionId = function() {
+      var maxId = 0;
+      this._misconceptions.forEach(function(misconception) {
+        var parsedIdAsInt = parseInt(misconception.getId(), 10);
+        maxId = Math.max(maxId, parsedIdAsInt);
+      });
+      var nextIdAsInt = maxId + 1;
+      return nextIdAsInt.toString();
+    };
+
+    Skill.prototype.findMisconceptionById = function(id) {
+      for (var idx in this._misconceptions) {
+        if (this._misconceptions[idx].getId() === id) {
+          return this._misconceptions[idx];
+        }
+      }
+      return null;
```

Comment:
I think it's fine to throw an error here.

Reply:
Done.